### PR TITLE
Checkstyle: Fix constant name violations in Matches (part 5)

### PR DESCRIPTION
**Pull request by ssoloff**
This PR fixes violations of the Checkstyle ConstantName rule in the `Matches` class by converting PascalCase fields to camelCase methods.

This is one part in a series of related PRs in order to keep the review size reasonable.

### CODE REVIEW

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 1]
```diff
@@ -1623,19 +1631,21 @@ private Matches() {}
...
-      return false;
-    }
-    return !ua.getRepairsUnits().isEmpty();
-  });
+  static Match<Unit> unitCanRepairOthers() {
+    return Match.of(unit -> {
+      if (unitIsDisabled().match(unit)) {
+        return false;
+      }
+      if (Matches.unitIsBeingTransported().match(unit)) {
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: Those 2 if statements can be merged into a single one.

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 2]
```diff
@@ -1694,23 +1704,25 @@ private Matches() {}
...
-    final UnitAttachment ua = UnitAttachment.get(unit.getType());
-    if (ua == null) {
-      return false;
-    }
-    return ua.getGivesMovement().size() > 0 && Matches.unitIsBeingTransported().invert().match(unit);
-  });
+  private static Match<Unit> unitCanGiveBonusMovement() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: This check could be merged with the return statement: `return ua != null && ua.getGivesMovement().size() > 0 && Matches.unitIsBeingTransported().invert().match(unit);`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 3]
```diff
@@ -1749,41 +1761,52 @@ private Matches() {}
...
-    }
-    return (ua.getCreatesUnitsList() != null && ua.getCreatesUnitsList().size() > 0);
-  });
+  static Match<Unit> unitCreatesUnits() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getCreatesUnitsList() != null && ua.getCreatesUnitsList().size() > 0;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: The if check can be merged with this return statement: `return ua != null && ua.getCreatesUnitsList() != null && ua.getCreatesUnitsList().size() > 0;`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 4]
```diff
@@ -1749,41 +1761,52 @@ private Matches() {}
...
-    }
-    return (ua.getCreatesResourcesList() != null && ua.getCreatesResourcesList().size() > 0);
-  });
+  static Match<Unit> unitCreatesResources() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getCreatesResourcesList() != null && ua.getCreatesResourcesList().size() > 0;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: The if check can be merged with this return statement: `return ua != null && ua.getCreatesResourcesList() != null && ua.getCreatesResourcesList().size() > 0;`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 5]
```diff
@@ -1749,41 +1761,52 @@ private Matches() {}
...
+  /**
+   * Returns a match indicating the specified unit type consumes at least one type of unit upon creation.
+   */
+  public static Match<UnitType> unitTypeConsumesUnitsOnCreation() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit);
+      if (ua == null) {
+        return false;
+      }
+      return ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: The if check can be merged with this return statement: `return ua != null && ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 6]
```diff
@@ -1749,41 +1761,52 @@ private Matches() {}
...
-    }
-    return (ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0);
-  });
+  static Match<Unit> unitConsumesUnitsOnCreation() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: The if check can be merged with this return statement: `return ua != null && ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 7]
```diff
@@ -1811,18 +1834,23 @@ private Matches() {}
...
+  /**
+   * Returns a match indicating the specified unit requires at least one type of unit upon creation.
+   */
+  public static Match<Unit> unitRequiresUnitsOnCreation() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getRequiresUnits() != null && ua.getRequiresUnits().size() > 0;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: The if check can be merged with this return statement: `return ua != null && ua.getRequiresUnits() != null && ua.getRequiresUnits().size() > 0;`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 8]
```diff
@@ -1856,25 +1884,34 @@ private Matches() {}
...
-
-  public static final Match<UnitType> UnitTypeIsConstruction = Match.of(type -> {
-    final UnitAttachment ua = UnitAttachment.get(type);
-    if (ua == null) {
+  static Match<Territory> territoryIsBlockadeZone() {
+    return Match.of(t -> {
+      final TerritoryAttachment ta = TerritoryAttachment.get(t);
+      if (ta != null) {
+        return ta.getBlockadeZone();
+      }
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: This if clause can be merged into a single return statement: `return ta != null && ta.getBlockadeZone();`

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 9]
```diff
@@ -1856,25 +1884,34 @@ private Matches() {}
...
+  /**
+   * Returns a match indicating the specified unit type is a construction unit type.
+   */
+  public static Match<UnitType> unitTypeIsConstruction() {
+    return Match.of(type -> {
+      final UnitAttachment ua = UnitAttachment.get(type);
+      if (ua == null) {
+        return false;
+      }
+      return ua.getIsConstruction();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: This return statement could be merged with the if statement: `return ua != null && ua.getIsConstruction();`

**Comment by ssoloff** [comment]
@RoiEXLab Please double-check my changes in the latest commit, as there is no unit test coverage here.

**Pull request merged**